### PR TITLE
Move to python3

### PR DESCRIPTION
**Pull request by juliushaertl**

### CODE REVIEW

File: .github/workflows/sphinxbuild.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: "Pull Request Docs Check"
```
Comment (skjnldsv) [inline .github/workflows/sphinxbuild.yml]: Awesome!!

Comment (ChristophWurst) [reply .github/workflows/sphinxbuild.yml]: lol I came here to say we should switch to gh actions once we're on python3 :rocket:

File: .github/workflows/sphinxbuild.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+on:
+- pull_request
```
Comment (J0WI) [inline .github/workflows/sphinxbuild.yml]:
```suggestion
- pull_request
- push
```

**Comment by ChristophWurst** [comment]
Rebased & squashed.

**Comment by ChristophWurst** [comment]
Adjusted the repo settings to the new required checks

**Pull request merged**